example/randomStrategies/src/randomStrategies.cpp

```diff
@@ -0,0 +1,220 @@
+    using QueueProperty = alpaka::Blocking;
+    using QueueAcc = alpaka::Queue<Acc, QueueProperty>;
+    using WorkDiv = alpaka::WorkDivMembers<Dim, Idx>;
+
+    QueueAcc queue; ///< default accelerator queue
+
+    // buffers holding the PRNG states
+    using BufHostRand = alpaka::Buf<Host, alpaka::rand::Philox4x32x10<Acc>, Dim, Idx>;
+    using BufAccRand = alpaka::Buf<Acc, alpaka::rand::Philox4x32x10<Acc>, Dim, Idx>;
+
+    Vec const extentRand; ///< size of the buffer of PRNG states
+    WorkDiv workdivRand; ///< work division for PRNG buffer initialization
+    BufHostRand bufHostRand; ///< host side PRNG states buffer (can be used to check the state of the states)
+    BufAccRand bufAccRand; ///< device side PRNG states buffer
+
+    // buffers holding the "simulation" results
+    using BufHost = alpaka::Buf<Host, float, Dim, Idx>;
+    using BufAcc = alpaka::Buf<Acc, float, Dim, Idx>;
+
+    Vec const extentResult; ///< size of the results buffer
+    WorkDiv workdivResult; ///< work division of the result calculation
+    BufHost bufHostResult; ///< host side results buffer
+    BufAcc bufAccResult; ///< device side results buffer
+
+    Box()
+        : queue{alpaka::getDevByIdx<Acc>(0u)}
+        , extentRand{static_cast<Idx>(NUM_POINTS)} // One PRNG state per "point".
+        , workdivRand{alpaka::getValidWorkDiv<Acc>(
+              alpaka::getDevByIdx<Acc>(0u),
+              extentRand,
+              Vec(1ul),
+              false,
+              alpaka::GridBlockExtentSubDivRestrictions::Unrestricted)}
+        , bufHostRand{alpaka::allocBuf<alpaka::rand::Philox4x32x10<Acc>, Idx>(
+              alpaka::getDevByIdx<Host>(0u),
+              extentRand)}
+        , bufAccRand{alpaka::allocBuf<alpaka::rand::Philox4x32x10<Acc>, Idx>(alpaka::getDevByIdx<Acc>(0u), extentRand)}
+        , extentResult{static_cast<Idx>((NUM_POINTS * NUM_ROLLS))} // Store all "rolls" for each "point"
+        , workdivResult{alpaka::getValidWorkDiv<Acc>(
+              alpaka::getDevByIdx<Acc>(0u),
+              extentResult,
+              Vec(static_cast<Idx>(NUM_ROLLS)), // One thread per "point"; each performs NUM_ROLLS "rolls"
+              false,
+              alpaka::GridBlockExtentSubDivRestrictions::Unrestricted)}
+        , bufHostResult{alpaka::allocBuf<float, Idx>(alpaka::getDevByIdx<Host>(0u), extentResult)}
+        , bufAccResult{alpaka::allocBuf<float, Idx>(alpaka::getDevByIdx<Acc>(0u), extentResult)}
+    {
+    }
+};
+
+/// Save the results file in CSV format. One line for each "point"; line length is the number of "rolls".
+void saveToFile(std::string filename, float const* buffer, Box const& box)
+{
+    std::ofstream output;
+    output.open(filename, std::ios::out);
+    std::cout << "Writing " << filename << std::endl;
+    auto lineLength = box.extentResult[0] / box.extentRand[0];
+    for(Box::Idx i = 0; i < box.extentResult[0]; ++i)
+    {
+        output << buffer[i] << ((i + 1) % lineLength ? ", " : "\n");
```

Comment:
I saw people writing TSV (tab separated values) a while back at CERN. That was unfamiliar at first, but i nicely eliminates the problem with locales, where e.g. en_US OS uses ',' as separator und '.' as decimal point, while 'de_DE` uses ';' as separater because it uses ',' as decimal point. So on a computer with a German OS, you will write a wrong file and have troubles opening these files in e.g. Excel. Thus I prefer TSV files in the meantime, they eliminate one of the two problems.

Reply:
A bit of googling just showed me that in Russian and Czechia `,` is also used as decimal separator. Might be relevant for some team members :)

Reply:
Yeah. In Czechia, we're used to using '.' as decimal separator while programming or doing data analysis. Commas are too much trouble so you usually expect your files to have a decimal point (unless it comes from MS Excel). I'll switch to TSV for now.